x-pack/legacy/plugins/siem/public/components/navigation/tab_navigation/index.tsx

```diff
@@ -36,71 +35,51 @@ const TabContainer = styled.div`
 
 TabContainer.displayName = 'TabContainer';
 
-interface TabNavigationState {
-  selectedTabId: string;
-}
-
-export class TabNavigation extends React.PureComponent<TabNavigationProps, TabNavigationState> {
-  constructor(props: TabNavigationProps) {
-    super(props);
-    const selectedTabId = this.mapLocationToTab(props.pageName, props.tabName);
-    this.state = { selectedTabId };
-  }
-  public componentWillReceiveProps(nextProps: TabNavigationProps): void {
-    const selectedTabId = this.mapLocationToTab(nextProps.pageName, nextProps.tabName);
-
-    if (this.state.selectedTabId !== selectedTabId) {
-      this.setState(prevState => ({
-        ...prevState,
-        selectedTabId,
-      }));
-    }
-  }
-  public render() {
-    const { display = 'condensed' } = this.props;
-    return (
-      <EuiTabs display={display} size="m">
-        {this.renderTabs()}
-      </EuiTabs>
-    );
-  }
-
-  public mapLocationToTab = (pageName: string, tabName?: HostsTableType): string => {
-    const { navTabs } = this.props;
+export const TabNavigation = React.memo<TabNavigationProps>(props => {
+  const { display = 'condensed', navTabs, pageName, showBorder, tabName } = props;
+  const mapLocationToTab = (): string => {
     return getOr(
       '',
       'id',
       Object.values(navTabs).find(item => tabName === item.id || pageName === item.id)
     );
   };
+  const [selectedTabId, setSelectedTabId] = useState<string>(mapLocationToTab());
+  useEffect(() => {
+    const currentTabSelected = mapLocationToTab();
 
-  private renderTabs = (): JSX.Element[] => {
-    const { navTabs } = this.props;
-    return Object.keys(navTabs).map(tabName => {
-      const tab = get(tabName, navTabs);
-      return (
-        <TabContainer
-          className={classnames({ euiTab: true, showBorder: this.props.showBorder })}
-          key={`navigation-${tab.id}`}
+    if (currentTabSelected !== selectedTabId) {
+      setSelectedTabId(currentTabSelected);
+    }
+  }, [pageName, tabName]);
```

Comment:
I am wondering if we should add `navTabs` in the dependency.

Reply:
I don't think we expect `navTabs` to change do we? previously, the `componentWillReceiveProps` only looked at `pageName` and `tabName` as those things update. but `navTabs` is the list with tab items that change which is active. does that make sense?